c2rust-transpile/src/c_ast/mod.rs

```diff
@@ -291,15 +290,15 @@ impl TypedAstContext {
     /// the targets of pointers
     pub fn is_forward_declared_type(&self, typ: CTypeId) -> bool {
         match self.resolve_type(typ).kind.as_underlying_decl() {
-            Some(decl_id) => match self[decl_id].kind {
-                CDeclKind::Struct { fields: None, .. } => true,
-                CDeclKind::Union { fields: None, .. } => true,
-                CDeclKind::Enum {
-                    integral_type: None,
-                    ..
-                } => true,
-                _ => false,
-            },
+            Some(decl_id) => matches!(
+                self[decl_id].kind,
+                CDeclKind::Struct { fields: None, .. }
+                    | CDeclKind::Union { fields: None, .. }
+                    | CDeclKind::Enum {
+                        integral_type: None,
+                        ..
+                    }
+            ),
```

Comment:
Fixed by https://github.com/immunant/c2rust/pull/460/commits/99be3a65c87c774c8d27872a9ed295f5605909d9.

Reply:
Personally I prefer `matches!` for code which can be expressed using it. One benefit is it is more concise. Even if it is formatted like in this PR, there are still less tokens obscuring the expression. The bigger benefit is that is a significantly weaker construct that a match: it always returns a `bool`, always does just a simple match and cannot include blocks as branches. I prefer to apply strength reduction when it doesn't compromise other code qualities.

Reply:
I generally don't like `matches!` for large patterns because it puts a pattern into non-first parameter position, which looks very un-idiomatic compared to other Rust code.

Still, I think it's a good idea to use `matches!` in places like these because of the strength reduction.  But if the code could be refactored to make it simpler elsewhere as well and in doing so eliminate a large `matches!`, I tend to prefer that.